From 2d09bcf4e4abaac6fc66c802da249619f55286a3 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Fri, 31 Mar 2023 20:17:00 +0200
Subject: [PATCH 0107/1141] arm64: dts: st: add st,axi-max-burst-len to dma
 controllers on stm32mp251

On stm32mp25x, the DMA3 instances are interconnected to AXI3 bus. AXI3
protocol supports a maximum burst length of 16 beats.
Use st,axi-max-burst-len to dma controllers nodes to limit burst length to
16 beats.

Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: I35e817ed49c08f79a3682cbe8349165b9761f4d4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/303304
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Valentin CARON <valentin.caron@foss.st.com>
Reviewed-by: Valentin CARON <valentin.caron@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 8ae33779e6b8..efac3e0735b9 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -173,6 +173,7 @@ hpdma: dma-controller@40400000 {
 				     <GIC_SPI 48 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&scmi_clk CK_SCMI_HPDMA1>;
 			#dma-cells = <4>;
+			st,axi-max-burst-len = <16>;
 		};
 
 		hpdma2: dma-controller@40410000 {
@@ -196,6 +197,7 @@ hpdma2: dma-controller@40410000 {
 				     <GIC_SPI 64 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&scmi_clk CK_SCMI_HPDMA2>;
 			#dma-cells = <4>;
+			st,axi-max-burst-len = <16>;
 		};
 
 		hpdma3: dma-controller@40420000 {
@@ -219,6 +221,7 @@ hpdma3: dma-controller@40420000 {
 				     <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&scmi_clk CK_SCMI_HPDMA3>;
 			#dma-cells = <4>;
+			st,axi-max-burst-len = <16>;
 		};
 
 		rifsc: rifsc@42080000 {
-- 
2.39.2

